File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.py
def solution(myString, pat):
    answer = 0
    a = []
    for i in myString:
        if i == "A":
            a.append("B")
        elif i == "B":
            a.append("A")
        nu_a = ''.join(a)
    if pat in nu_a:
        answer = 1
    else:
        answer = 0
    return answer

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        StringBuilder sb = new StringBuilder();
        for (char c : myString.toCharArray()) {
            sb.append(c == 'A' ? 'B' : 'A');
        }
        String newString = sb.toString();
        
        return newString.contains(pat) ? 1 : 0;
    }
}